router/batchrouter/batchrouter.go

```diff
@@ -2410,3 +2412,25 @@ func (brt *HandleT) updateProcessedEventsMetrics(statusList []*jobsdb.JobStatusT
 		}).Count(count)
 	}
 }
+
+func tryUnmarshalJSON(jobID int64, data []byte, v interface{}) (err error) {
+	c := cap(data)
+	l := len(data)
+
+	startingData := make([]byte, l)
+	copy(startingData, data)
+
+	defer func() {
+		if r := recover(); r != nil {
+			pkgLogger.Warnf(
```

Comment:
How are we going to be alerted and make sure this piece of code won't be left there forever, in oblivion?

Reply:
We can do this: https://github.com/rudderlabs/rudder-devops/blob/e64392172c6beddb7991adcc5907435806177853/helm-charts/loki-stack/values.infra1-us.yaml#L73

Reply:
Since this issue is not frequent (2 occurences so far), it might be easier for us to detect and debug if we didn't shallow the panic, but simply log some more stuff, either in the panic message itself or in the logs.